pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -323,11 +326,15 @@ func syncVolume(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCl
 	case api.VolumeReleased:
 		if volume.Spec.ClaimRef == nil {
 			return fmt.Errorf("PersistentVolume[%s] expected to be bound but found nil claimRef: %+v", volume.Name, volume)
-		} else {
-			// another process is watching for released volumes.
-			// PersistentVolumeReclaimPolicy is set per PersistentVolume
-			//  Recycle - sets the PV to Pending and back under this controller's management
-			//  Delete - delete events are handled by this controller's watch. PVs are removed from the index.
+		}
+		// another process is watching for released volumes.
+		// PersistentVolumeReclaimPolicy is set per PersistentVolume
+		//  Recycle - sets the PV to Pending and back under this controller's management
+		//  Delete - delete events are handled by this controller's watch. PVs are removed from the index.
+		_, err := binderClient.GetPersistentVolumeClaim(volume.Spec.ClaimRef.Namespace, volume.Spec.ClaimRef.Name)
```

Comment:
This is wrong. Volumes in Released phase **must** be recycled or deleted first before binding to another claim, even if it has the same namespace+name!

We got bitten by this in several demos - you create PVC and start a pod with say MySQL, show that your app is running and you can add some data to your database. Now, when you delete your pod and PVC and create it again (from the same yaml files), you must get fresh MySQL with no data, not the old one.